safe-global/safe-wallet-web
# feat: recovery queue

## What it solves

Resolves #2763

## How this PR fixes it

This implements a new "Pending recovery" section to the transaction queue which displays all recovery attempts from connected recovery-specific Delay Modifiers, as well as associated adjustments:

- Recovery queue with decoded overview
- Global state adjustments (`executor` and `isMalicious` flags have been added, as well as all timestamps converted to ms)
- Associated unit tests

Note: text was tweaked and does not necessarily match the design. It is also not possible to skip recovery proposals yet. As such, queued recovery attempts and their "skip" transactions are not included in this PR.

## How to test it

1. Propose a recovery attempt
2. Observe recovery proposal in queue, displaying cooldown, expiration and option to execute

## Screenshots

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/97d8a9da-a1f2-41eb-8ffe-4e36eaa77fa5)

Old naming/colouring:
![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/f8c3598a-feaf-4321-97d8-fba9573e1789)

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/0f6d435b-9657-4569-8328-348ef4ded05c)

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/36a24a15-a150-47f8-afd6-035a17eaba6e)

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/24df79ad-7006-4aed-9771-032466515d91)

## Checklist
